hera_cal/smooth_cal.py

```diff
@@ -79,34 +79,32 @@ def freq_filter(gains, wgts, freqs, filter_scale=10.0, tol=1e-09, window='tukey'
         filter_scale: frequency scale in MHz to use for the low-pass filter. filter_scale^-1 corresponds
             to the half-width (i.e. the width of the positive part) of the region in fourier
             space, symmetric about 0, that is filtered out.
-        tol: CLEAN algorithm convergence tolerance (see aipy.deconv.clean)
-        window: window function for filtering applied to the frequency axis.
-            See aipy.dsp.gen_window for options.
         skip_wgt: skips filtering rows with very low total weight (unflagged fraction ~< skip_wgt).
             filtered is left unchanged and info is {'skipped': True} for that time.
             Only works properly when all weights are all between 0 and 1.
-        maxiter: Maximum number of iterations for aipy.deconv.clean to converge.
-        win_kwargs : any keyword arguments for the window function selection in aipy.dsp.gen_window.
+        mode: deconvolution method to use. See uvtools.dspec.fourier_filter for full list of supported modes.
+              examples include 'dpss_leastsq', 'clean'.
+        filter_kwargs : any keyword arguments for the window function selection in aipy.dsp.gen_window.
             Currently, the only window that takes a kwarg is the tukey window with a alpha=0.5 default.
-
     Returns:
         filtered: filtered gains, ndarray of shape=(Ntimes,Nfreqs)
         info: info object from uvtools.dspec.high_pass_fourier_filter
     '''
     if not HAVE_UVTOOLS:
         raise ImportError("uvtools required, instsall hera_cal[all]")
 
-    df = np.median(np.diff(freqs))  # in Hz
     filter_size = (filter_scale * 1e6)**-1  # Puts it in s
     dly = single_iterative_fft_dly(gains, wgts, freqs)  # dly in s
     rephasor = np.exp(-2.0j * np.pi * dly * freqs)
-    filtered, res, info = uvtools.dspec.high_pass_fourier_filter(gains * rephasor, wgts, filter_size, df, tol=tol, window=window,
-                                                                 skip_wgt=skip_wgt, maxiter=maxiter, **win_kwargs)
-    filtered /= rephasor
+
+    filtered, res, info = uvtools.dspec.fourier_filter(x=freqs, data=gains * rephasor, wgts=wgts, mode=mode, filter_centers=[0.],
```

Comment:
Since this now supports a filter center other than zero, do you want to try removing the rephasor in favor of a using the `dly` as the filter center?

Reply:
I think that would be more elegant.

Reply:
I've changed to filter_centers=[dly] and gotten rid of the rephasoring.

Reply:
This actually passes unit tests for `dpss` fitting but not for `clean`. Potentially because of the discretized clean area. Rather then dive into this rabbit hole, I'm going to keep things as is.